src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -321,25 +303,19 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
             }
 
             await ReloadLoadedPackagesAsync();
-
-            AfterLoadUnload(package);
+            await AfterLoadUnloadAsync(package);
         }
 
-        private void BeforeLoadUnload() {
+        private void BeforeLoadUnload(IRPackageViewModel package) {
             if (_selectedTab == SelectedTab.InstalledPackages || _selectedTab == SelectedTab.LoadedPackages) {
                 IsLoading = true;
             }
+            package.IsChanging = true;
         }
 
-        private void AfterLoadUnload(IRPackageViewModel package) {
+        private async Task AfterLoadUnloadAsync(IRPackageViewModel package) {
+            await ReplaceItemsAsync();
```

Comment:
It checks which tab is current, while we need to check if current tab is the one that was updated. If while updating user has switched the tab, we shouldn't call `ReplaceItemsAsync`, otherwise we will populate the list with outdated items.

---
In reply to: [73554727](https://github.com/Microsoft/RTVS/pull/2149#discussion_r73554727) [](ancestors = 73554727)